.github/workflows/validate-pr.yml

```diff
@@ -4,6 +4,15 @@ on:
     types: [edited, opened, reopened, synchronize]
 
 jobs:
+  # We want to prevent cross-contamination between the 3.x and 4.x pipelines. So we should prevent PRs
+  # based on this flow to merge into `dev-3` or `release-3`.
+  verify_target_branch:
+    runs-on: ubuntu-latest
+    steps:
+      - if: ${{ github.base_ref == 'dev-3' || github.base_ref == 'release-3' }}
+        run: |
+          echo "Forbidden to merge this branch into dev-3 or release-3"
+          exit 1
```

Comment:
A neat trick is to use :
   echo "::error Forbidden to merge this branch into dev-3 or release-3"
instead of using exit 1 so that the error shows up on the summary page.

Reply:
Oh, that _is_ neat. Thanks!